10 спринт

IvanPodlesnykh opened the pull request.

IvanPodlesnykh committed:
Приложение Playlist Maker
Задание за 10 спринт:
1. Добавил RecyclerView на экран поиска
2. Реализовал классы Track, TrackViewHolder и TrackAdapter для отображения треков в RecyclerView
3. Реализовал верстку элементов RecyclerView для треков
4. Обложки треков подружаются из интернета с помощью библиотеки Glide

IvanPodlesnykh committed:
Приложение Playlist Maker
Задание за 10 спринт:
Вынес размеры в dimens
Убрал пример названия трека, исполнителя и время из layout файла

Nick-review commented on app/src/main/java/com/ivanpodlesnykh/playlistmaker/TrackViewHolder.kt:
```diff
@@ -0,0 +1,31 @@
+        val songName = itemView.findViewById<TextView>(R.id.song_name)
+        val artistName = itemView.findViewById<TextView>(R.id.artist_name)
+        val trackTime = itemView.findViewById<TextView>(R.id.track_time)
+        val trackCover = itemView.findViewById<ImageView>(R.id.track_cover)
```
Все переменные с присвоениями вью переменным необходимо вынести в тело класса `TrackViewHolder`
метод bind у recycler view может вызываться очень часто, нет смысла там каждый раз создавать переменные, если можно сделать это 1 раз в теле класса
И общее правило работы такое, что сложные операции в методе bind лучше не выполнять, это может тормозить список и он будет лагать при скроле. К этой ситуации это не относится, на будущее

Nick-review commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -58,5 +58,11 @@
+        android:id="@+id/list_of_tracks"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:paddingTop="16dp"
```
Здесь и во всё классе необходимо вынести в `dimens.xml`

Nick-review commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -58,5 +58,11 @@
             android:paddingRight="12dp"
```
Здесь и во всё классе необходимо вынести в `dimens.xml`

Nick-review commented on app/src/main/res/layout/track_item.xml:
```diff
@@ -0,0 +1,57 @@
+            android:layout_height="@dimen/track_name_height"
+            style="@style/TrackName" />
+        <LinearLayout
+            android:layout_width="250dp"
```
Здесь и во всё классе необходимо вынести в `dimens.xml`

Nick-review replied:
![Screenshot_3](https://github.com/IvanPodlesnykh/PlaylistMaker/assets/151505123/63923406-109c-4dac-8b3f-4f8e9615069f)
Из-за того, что размер у вас прибит "гвоздями", имя артиста не помещается полностью, хотя места справа достаточно, чтобы отрисовать имя полностью. Рекомендую вообще весь экран сделать при помощи ConstraintLayout. Это уберёт излишнюю вложенность и такую вёрстку будет сложнее сломать.

IvanPodlesnykh committed:
Приложение Playlist Maker
Правки 10 спринт:
1. Вынес все размеры в dimens
2. Изменил layout track_item на constraint
3. Названия занимают свободное пространство и обрезаются, если превышают его
4. Вынес переменные, хранящие View-элементы track_item, из метода bind в класс

Nick-review commented on app/src/main/res/layout/track_item.xml:
```diff
@@ -1,57 +1,69 @@
 <?xml version="1.0" encoding="utf-8"?>
-
 <LinearLayout
```
Этот `LinearLayout `можно убрать полностью, есть `Constraint`, в данном случае он будет корневой `ViewGroup`

Nick-review commented on app/src/main/res/layout/track_item.xml:
```diff
@@ -1,57 +1,69 @@
+        android:id="@+id/track_time"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        app:layout_constraintStart_toEndOf="@+id/dot"
```
Вот здесь было бы неплохо ещё сделать привязку End-to-Start к стрелочке, как это сделано у названия трека
Сейчас если вставить слишком длинное время, то оно уходить за рамки экрана
Это не критическое замечание, скорее всего такая ситуация никогда не произойдёт, что придёт такое длинное время, но может быть полезно в будущем, когда будете создавать другие экраны
![Screenshot_1](https://github.com/IvanPodlesnykh/PlaylistMaker/assets/151505123/999a75d4-d317-4e77-af7f-a1116aa192af)
